src/Build/BackEnd/Shared/BuildRequestConfiguration.cs

```diff
@@ -1024,23 +1030,17 @@ private static string ResolveToolsVersion(BuildRequestData data, string defaultT
         /// <summary>
         /// Gets the translator for this configuration.
         /// </summary>
-        private ITranslator GetConfigurationTranslator(TranslationDirection direction)
+        private ITranslator GetConfigurationTranslator(TranslationDirection direction, Stream stream, string cacheFile)
         {
-            string cacheFile = GetCacheFile();
             try
             {
-                if (direction == TranslationDirection.WriteToStream)
-                {
-                    Directory.CreateDirectory(Path.GetDirectoryName(cacheFile));
-                    return BinaryTranslator.GetWriteTranslator(File.Create(cacheFile));
-                }
-                else
-                {
+                return direction == TranslationDirection.WriteToStream
+                    ? BinaryTranslator.GetWriteTranslator(stream)
+
                     // Not using sharedReadBuffer because this is not a memory stream and so the buffer won't be used anyway.
-                    return BinaryTranslator.GetReadTranslator(File.OpenRead(cacheFile), InterningBinaryReader.PoolingBuffer);
-                }
+                    : BinaryTranslator.GetReadTranslator(stream, InterningBinaryReader.PoolingBuffer);
             }
-            catch (Exception e) when (e is DirectoryNotFoundException || e is UnauthorizedAccessException)
+            catch (Exception e) when (e is DirectoryNotFoundException or UnauthorizedAccessException)
             {
                 ErrorUtilities.ThrowInvalidOperation("CacheFileInaccessible", cacheFile, e);
```

Comment:
We now open the file outside of the `try` block so this will probably not work anymore.